### DISCUSSION TITLE
Kebord shortcut 'i' inverting side panel colors

### DISCUSSION THREAD

**Discussion by crustyturd** [sumatrapdfreader/sumatrapdf#2552] [General]
I may be wrong, but I think previous pre-releases inverted colors only for document panel.
With currrent pre-release colors are inverted for bookmarks, favorites as well. Was this intentional?
Because I much preferred  inverting colors for document panel only.
It would be even better if it was possible to invert images only.

**Comment by GitHubRulesOK** [comment]
Anytime (and yes there were occasions) the side pane(s) colors were out of sync with canvas body, its raised as a 🐛 "bug" 
the current short cut to invert all colors (including side panes) ***should be*** `Shift+ I` (so currently not right? ?)
and in most cases collateral damage is both inverted images and unseen text at the side.

**Reply by crustyturd** [reply]
So, if I  understand, the previous version that did not invert side pane color was a bug?  That's O.K. But  I  liked it better when only the canvas body was inverted as I'm really only interested in temporarily inverting the images anyway. But I can live with it.
Yes, 'I' currently inverts colors and 'Shift + I' shows page info.

**Comment by kjk** [comment]
There are 2 separate (but also related) concerns:
* theming of the UI (i.e. changing colors), mostly to get dark theme
* inverting the colors of the rendered page

We don't have theming (because getting all the details right is difficult because theming windows controls is hard).

As you see, `'i` currently does a little of both because more people seem to prefer it over not doing that.

We can't really theme PDF rendering because the color of every pixel is defined in PDF file so you can't just say "change background to this color and font color to that color". Inverting colors is a bit of a hack to get "white text on dark background" rendering.

If we had themes, whether inverting colors should be done automatically as part of dark theme is an open question. I guess it should be controlled independently i.e. with theming, `i` would go back to only toggling page color inversions and there would be advanced settings option to specify if inverting colors is part of dark theme or not.
